packages/core/parcel-bundler/src/assets/KotlinAsset.js

```diff
@@ -0,0 +1,65 @@
+const Asset = require('../Asset');
+const localRequire = require('../utils/localRequire');
+
+class KotlinAsset extends Asset {
+  constructor(name, options) {
+    super(name, options);
+    this.type = 'js';
+  }
+
+  async generate() {
+    // require kotlin
+    const kotlinCompiler = await localRequire(
+      '@jetbrains/kotlinc-js-api',
+      this.name
+    );
+    const fs = require('fs', this.name);
+
+    // remove extension and path
+    let slashyIndex = this.id.lastIndexOf('/'); // linux/mac
+    if (slashyIndex === -1) slashyIndex = this.id.lastIndexOf('\\'); // windows
+    const fileName = this.id.substring(slashyIndex, this.id.lastIndexOf('.'));
+
+    await kotlinCompiler.compile({
+      output: 'build/kt.temp/' + fileName + '.js',
```

Comment:
You should be explicit about exactly where this is going to go. Right now it will go into the current working directory. Instead, it would be better to go into a temporary directory.